core/grant/service.go

```diff
@@ -254,6 +256,34 @@ func (s *Service) Revoke(ctx context.Context, id, actor, reason string, opts ...
 	return grant, nil
 }
 
+func (s *Service) Restore(ctx context.Context, id, actor, reason string) (*domain.Grant, error) {
+	grant, err := s.GetByID(ctx, id)
+	if err != nil {
+		return nil, fmt.Errorf("getting grant details: %w", err)
+	}
+
+	if err := grant.Restore(actor, reason); err != nil {
+		return nil, fmt.Errorf("%w: %s", ErrInvalidRequest, err.Error())
```

Comment:
this will always make error from `grant.Restore` to be `ErrInvalidRequest`.
is that intended?

one alternative is to let `grant.Restore` decide what error to send and this line can just be

```suggestion
		return nil, fmt.Errorf("error restoring grant: %w", err)
```

Reply:
yes that is intended bcs `grant.Restore` here is only updating grant record in memory. The approach you suggested will also work, both ways are acceptable IMO

Reply:
for now it is still okay, but one downside of handling `ErrInvalidRequest` here is if `grant.Restore` has more logic that may not define invalid request, then we are marking the error falsely.

right now this line heavily assumes that grant.Restore error will always be `ErrInvalidRequest`

Reply:
ah that's right, let me update